### PR TITLE
Support params as list for exploding parameters

### PR DESCRIPTION
**Pull request by xiangyan99**
#12410

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 1]
```diff
@@ -300,7 +300,14 @@ def format_parameters(self, params):
...
             params.update(existing_params)
-        query_params = ["{}={}".format(k, v) for k, v in params.items()]
+        query_params = []
+        for k, v in params.items():
+            if isinstance(v, str):
+                query_params.append("{}={}".format(k, v))
+            elif isinstance(v, list):
+                query_params.extend(["{}={}".format(k, w) for w in v])
+            else:
+                raise ValueError()
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:310]: Do we want to be that precise? Format could call automatically `__str__` on an object, with this code using integer will fail for example now.
What about:
```python
            if isinstance(v, list):
                query_params.extend(["{}={}".format(k, w) for w in v])
            else:
                query_params.append("{}={}".format(k, v))
```

**Comment by lmazuel** [comment]
@johanste autorest is modeling a `null` element in that array as "empty string", so sending `[None, '']` will send `a=,` or `a=&a=` depending of the multi/explode parameter. Thoughts? I don't think there is clear spec on what null would mean here

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py (hunk 1 above)
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_base.py:310]: Good catch. My implementation is a potential breaking change. Thanks.

**Comment by johanste** [comment]
@lmazuel - If you insist on being able to send `null` values in query strings, then it should probably be a url encoded `null` (%00).

**Comment by lmazuel** [comment]
@johanste I just want to be prepared :p. We need to do something, whatever this something is, and I just feel sending the string `None` s not what we want (I mean, I know it's not what we want right). But I'm not sure what we want, I'm happy to raise or use `%00` I have no strong opinions.

**Comment by lmazuel** [comment]
@johanste @xiangyan99 given we don't do any encoding in azure-core, I would raise a `ValueError` if you use a `None` value, up to you SDK developer to handle that in the way your server is expecting to deal with `null` semantic (%00, empty string or else)

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 2]
```diff
@@ -300,7 +300,15 @@ def format_parameters(self, params):
...
-        query_params = ["{}={}".format(k, v) for k, v in params.items()]
+        query_params = []
+        for k, v in params.items():
+            if isinstance(v, list):
+                for w in v:
+                    if w is None:
+                        raise ValueError()
+                    query_params.append("{}={}".format(k, w))
+            else:
+                query_params.append("{}={}".format(k, v))
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:311]: We should raise for `v` too here

**Pull request merged**